arangod/Agency/Supervision.cpp

```diff
@@ -2219,6 +2225,91 @@ void Supervision::checkBrokenAnalyzers() {
   }
 }
 
+
+void Supervision::checkReplicatedLogs() {
+  _lock.assertLockedByCurrentThread();
+
+  using namespace replication2::agency;
+
+  auto const readPlanSpecification = [](Node const& node) -> LogPlanSpecification {
+    // TODO hack
```

Comment:
Probably this is not so bad, at least it should work fine. Shall we remove the TODO here and in the following places?